src/main/java/flight/reservation/payment/CreditCard.java

```diff
@@ -1,31 +1,55 @@
 package flight.reservation.payment;
 
+import flight.reservation.order.FlightOrder;
+
 import java.util.Date;
 
 /**
  * Dummy credit card class.
  */
-public class CreditCard {
-    private double amount;
-    private String number;
-    private Date date;
-    private String cvv;
+public class CreditCard extends PaymentMethod {
+    private final String number;
+    private final Date date;
+    private final String cvv;
     private boolean valid;
 
     public CreditCard(String number, Date date, String cvv) {
-        this.amount = 100000;
+        super();
         this.number = number;
         this.date = date;
         this.cvv = cvv;
         this.setValid();
     }
 
-    public void setAmount(double amount) {
-        this.amount = amount;
+    public boolean processOrder(FlightOrder flightOrder) throws IllegalStateException {
+        if (flightOrder.isClosed()) {
+            // Payment is already proceeded
+            return true;
+        }
+        // validate payment information
+        if (!isValid()) {
+            throw new IllegalStateException("Payment information is not set or not valid.");
+        }
+        boolean isPaid = pay(flightOrder, flightOrder.getPrice());
+        if (isPaid) {
+            flightOrder.setClosed();
+        }
+        return isPaid;
     }
 
-    public double getAmount() {
-        return amount;
+    public boolean pay(FlightOrder flightOrder, double amount) throws IllegalStateException {
+        if (isValid()) {
+            System.out.println("Paying " + flightOrder.getPrice() + " using Credit Card.");
+            double remainingAmount = getAmount() - flightOrder.getPrice();
+            if (remainingAmount < 0) {
+                System.out.printf("Card limit reached - Balance: %f%n", remainingAmount);
+                throw new IllegalStateException("Card limit reached");
+            }
+            setAmount(remainingAmount);
```

Comment:
not sure if `public setAmount` is necessary? Here, you could directly operate on the `protected` member variable.